deps/build.jl

```diff
@@ -11,85 +11,65 @@ struct HSLVersion
   ext::String
 end
 
+include("versions.jl")
+
 getname(ver::HSLVersion) = ver.algname * "-" * ver.version * ver.ext
 
-function checksha(version::HSLVersion, filepath)
-  if isfile(filepath)
-    open(filepath) do f
+function checksha(version::HSLVersion, archivepath::String)
+  if isfile(archivepath)
+    open(archivepath) do f
       return bytes2hex(sha256(f)) == version.sha
     end
   else
     return false
   end
 end
 
-function findversion(versions::Vector{HSLVersion}, path::AbstractString)
-  for ver in versions
-    archivename = getname(ver)
-    archivepath = joinpath(path, archivename)
-    if isfile(archivepath)
-      if checksha(ver, archivepath)
-        return ver
-      else
-        error("Archive found but didn't match SHA for $(ver.algname).")
+"""
+Find a version of a software whose archive is in one of the search paths.
+Searches `downloads` folder, ENV["HSL_ARCHIVES_PATH"] and ENV["<ALGNAME>_PATH"], e.g.
+"HSL_MA97_PATH" or "MC21_PATH".
+"""
+function findversion(software::String)
+  paths = String[joinpath(@__DIR__, "downloads")]
+  haskey(ENV, "HSL_ARCHIVES_PATH") && push!(paths, ENV["HSL_ARCHIVES_PATH"])
+  haskey(ENV, "$(uppercase(software))_PATH") && push!(paths, ENV["$(uppercase(software))_PATH"])
+  versions = hsl_collection[software]
```

Comment:
We know that because `haskey(hsl_collection, software) == true` because I call the `findversion` function in the following loop:
```julia
for software in keys(hsl_collection)
  path, version, patchs = findversion(software)
  if version ≠ nothing
    hsl_versions[software] = version
    hsl_archives[software] = path
    if software == "hsl_ma57"
      hsl_patches[software] = patchs
    end
  end
end
```